tensorboard/main.py

```diff
@@ -35,20 +35,24 @@
 os.environ['GCS_READ_CACHE_DISABLED'] = '1'
 # pylint: enable=g-import-not-at-top
 
+import logging
 import sys
 
-# Import TensorFlow here to fail immediately if it's not present, even though we
-# don't actually use it yet, which results in a clearer error.
-import tensorflow as tf  # pylint: disable=unused-import
```

Comment:
I think we'd still like to ensure that TF is required by default for now, until we can add the improved UI messaging and more complete CI test coverage.  What if we made this import check bypassable with an env var?  (Like the flag idea, but simpler to implement here).

```python
import os

def run_main():
  ...
  if getattr(tf, '__version__', 'stub') == 'stub':
    # Unless the user has explicitly requested running without TensorFlow by setting the
    # TENSORBOARD_NO_TF environment variable, we check for TensorFlow here so that if it's
    # missing we generate a clear and immediate error rather than partial functionality.
    if os.getenv('TENSORBOARD_NO_TF') is None:
      import tensorflow  # pylint: disable=unused-import
    logger.warn("TensorFlow installation not found - running with reduced feature set.")
```

We could include a TODO with a GitHub issue to track the work needed to flip this to be on by default.

Reply:
I like it. I agree that the UI, given that it's still shown in the same form right now, needs to be updated. This unblocks this incremental step, though. Making the change.

Reply:
Created https://github.com/tensorflow/tensorboard/issues/2027, which I'll reference.

Reply:
Great, thanks!